crates/data_components/src/duckdb.rs

```diff
@@ -166,11 +175,36 @@ impl TableProviderFactory for DuckDBTableProviderFactory {
         let mode: Mode = mode.as_str().into();
 
         let indexes_option_str = options.remove("indexes");
-        let indexes = match indexes_option_str {
-            Some(indexes_str) => indexes::indexes_from_option_string(&indexes_str),
+        let unparsed_indexes: HashMap<String, IndexType> = match indexes_option_str {
+            Some(indexes_str) => util::hashmap_from_option_string(&indexes_str),
             None => HashMap::new(),
         };
 
+        let unparsed_indexes = unparsed_indexes
+            .into_iter()
+            .map(|(key, value)| {
+                let columns = ColumnReference::try_from(key.as_str())
+                    .context(UnableToParseColumnReferenceSnafu)
+                    .map_err(to_datafusion_error);
+                (columns, value)
+            })
+            .collect::<Vec<(Result<ColumnReference, DataFusionError>, IndexType)>>();
+
+        let mut indexes: Vec<(ColumnReference, IndexType)> = Vec::new();
+        for (columns, index_type) in unparsed_indexes {
+            let columns = columns?;
+            indexes.push((columns, index_type));
+        }
+
+        let mut on_conflict: Option<OnConflict> = None;
+        if let Some(on_conflict_str) = options.remove("on_conflict") {
```

Comment:
This is where we de-serialize the `on_conflict` string back into the `OnConflict` struct. I also tweaked the indexing de-serializing above to use the new ColumnReference type. In theory this de-serialize should never fail - but being defensive and returning an error if we can't.